core/eval.go

```diff
@@ -2314,6 +2314,51 @@ func evalPTTL(args []string, store *Store) []byte {
 	return Encode(int64(durationMs), false)
 }
 
+// evalHSET sets the specified fields to their
+// respective values in an hashmap stored at key
+//
+// This command overwrites the values of specified
+// fields that exist in the hash.
+//
+// If key doesn't exist, a new key holding a hash is created.
+//
+// Usage: HSET key field value [field value ...]
+func evalHSET(args []string, store *Store) []byte {
+	if len(args) < 3 {
+		return diceerrors.NewErrWithMessage("wrong number of arguments for command")
+	}
+
+	key := args[0]
+
+	obj := store.Get(key)
+
+	var hashMap HashMap
+	var numKeys int64
+
+	if obj != nil {
+		switch currentVal := obj.Value.(type) {
+		case HashMap:
+			hashMap = currentVal
+		default:
+			return diceerrors.NewErrWithMessage("key already exists")
```

Comment:
I believe if the key is already associated with another type, it should throw error as below
> (error) WRONGTYPE Operation against a key holding the wrong kind of value
We can do
 `return diceerrors.NewErrWithMessage(diceerrors.WrongTypeErr)`